ludwig/data/preprocessing.py

```diff
@@ -1348,8 +1357,13 @@ def _preprocess_file_for_training(
             logger.info('Writing preprocessed dataset cache')
             data_hdf5_fp = replace_file_extension(dataset, 'hdf5')
             data_utils.save_hdf5(data_hdf5_fp, data, training_set_metadata)
-            training_set_metadata[DATA_TRAIN_HDF5_FP] = data_hdf5_fp
+
             logger.info('Writing train set metadata')
+            training_set_metadata[DATA_TRAIN_HDF5_FP] = data_hdf5_fp
+            training_set_metadata[CHECKSUM] = calculate_checksum(
+                training_set,
```

Comment:
instead of specifying `training_set` should be `dataset` because this section of code is executed if `dataset=` parameter is specified on the train().